### PR TITLE
feat(webserver): add endpoint to calculate disk usage

### PR DESCRIPTION
**Pull request by boxbeam**
cc @wwayne 
Closes TAB-585

### CODE REVIEW

File: ee/tabby-webserver/src/service/analytic.rs [hunk 1]
```diff
@@ -70,6 +71,41 @@ impl AnalyticService for AnalyticServiceImpl {
...
+                .await?
+                .combine(recursive_dir_size(tabby_common::path::index_dir()).await?),
+            database: recursive_dir_size(crate::path::tabby_ee_root()).await?,
+            models: recursive_dir_size(tabby_common::path::models_dir()).await?,
+        })
+    }
+}
+
+/// Calculate the size of a directory in kilobytes recursively
+fn recursive_dir_size(
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/analytic.rs:88]: consider use https://docs.rs/fs_extra/1.2.0/fs_extra/dir/fn.get_size.html

File: ee/tabby-webserver/src/service/analytic.rs [hunk 2]
```diff
@@ -70,6 +71,41 @@ impl AnalyticService for AnalyticServiceImpl {
             .collect();
         Ok(stats)
     }
+
+    async fn storage_stats(&self) -> Result<StorageStats> {
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/analytic.rs:75]: add unit test coverage

File: ee/tabby-webserver/src/schema/analytic.rs [hunk 3]
```diff
@@ -8,6 +8,33 @@ use strum::{EnumIter, IntoEnumIterator};
 
 use crate::schema::Result;
 
+#[derive(GraphQLObject)]
+pub struct StorageStats {
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/schema/analytic.rs:12]:
```suggestion
pub struct DiskUsageStats {
```

File: ee/tabby-webserver/src/schema/analytic.rs [hunk 4]
```diff
@@ -8,6 +8,33 @@ use strum::{EnumIter, IntoEnumIterator};
...
+    pub events: DirectoryStat,
+    pub indexed_repositories: DirectoryStat,
+    pub database: DirectoryStat,
+    pub models: DirectoryStat,
+}
+
+#[derive(GraphQLObject)]
+pub struct DirectoryStat {
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/schema/analytic.rs:20]:
```suggestion
pub struct DiskUsage {
```

**Review by wsxiaoys — COMMENTED** [review]
Otherwise LG

File: ee/tabby-webserver/src/schema/mod.rs [hunk 5]
```diff
@@ -457,6 +457,12 @@ impl Query {
         )
         .await
     }
+
+    async fn storage_stats(ctx: &Context) -> Result<StorageStats> {
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/schema/mod.rs:461]:
```suggestion
    async fn disk_usage_stats(ctx: &Context) -> Result<StorageStats> {
```

File: ee/tabby-webserver/src/service/analytic.rs (hunk 1 above)
Comment (boxbeam) [reply ee/tabby-webserver/src/service/analytic.rs:88]: Not async?

Comment (wsxiaoys) [reply ee/tabby-webserver/src/service/analytic.rs:88]: use spawn blocking?

**Review by wsxiaoys — APPROVED** [review]

**Pull request merged**